x/concentrated-liquidity/tick.go

```diff
@@ -183,6 +184,85 @@ func GetMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne sdk.Int) (minTic
 	return math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
 }
 
+// GetTickLiquidityForRangeInBatches returns an array of liquidity depth within the given range of lower tick and upper tick.
+func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]query.LiquidityDepthWithRange, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// sanity check that pool exists and upper tick is greater than lower tick
+	if !k.poolExists(ctx, poolId) {
+		return []query.LiquidityDepthWithRange{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+
+	// use false for zeroForOne since we're going from lower tick -> upper tick
+	zeroForOne := false
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey, sdk.ZeroDec())
+
+	// get min and max tick for the pool
+	p, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return []query.LiquidityDepthWithRange{}, err
+	}
+	exponentAtPriceOne := p.GetPrecisionFactorAtPriceOne()
+	minTick, maxTick := math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
+
+	// set current tick to min tick, and find the first initialized tick starting from min tick -1.
+	// we do -1 to make min tick inclusive.
+	currentTick := minTick - 1
+
+	nextTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick)
+	if !ok {
+		return []query.LiquidityDepthWithRange{}, types.InvalidTickError{Tick: currentTick, IsLower: false, MinTick: minTick, MaxTick: maxTick}
+	}
+
+	tick, err := k.GetTickByTickIndex(ctx, poolId, nextTick)
+	if err != nil {
+		return []query.LiquidityDepthWithRange{}, err
+	}
+
+	liquidityDepthsForRange := []query.LiquidityDepthWithRange{}
+
+	// use the smallest tick initialized as the starting point for calculating liquidity.
+	currentLiquidity := tick.LiquidityNet
+	currentTick = nextTick.Int64()
+
+	totalLiquidityWithinRange := currentLiquidity
+	for currentTick <= maxTick {
+		nextTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick)
+		// break and return the liquidity as is if
+		// - there are no more next tick that is initialized,
+		// - we hit upper limit
+		if !ok {
+			break
+		}
+
+		keyTick := types.KeyTick(poolId, nextTick.Int64())
+		tickStruct := model.TickInfo{}
+		found, err := osmoutils.Get(store, keyTick, &tickStruct)
```

Comment:
Can we use the `GetTickByTickIndex` here?

Reply:
Changed :)